docs/enhancements/image-service-cloud-authentication.md

```diff
@@ -0,0 +1,151 @@
+---
+title: image-service-cloud-authentication
+authors:
+  - "@carbonin"
+creation-date: 2021-10-07
+last-updated: 2021-10-07
+---
+
+# Image Service Cloud Authentication
+
+## Summary
+
+Authentication between the image service and assisted service when deployed
+in the cloud requires a new strategy because the existing tokens used in the cloud
+are too large and too short lived to be included in a pre-signed URL. Instead
+of the existing authentication used in the cloud, assisted service will issue JWTs
+using a symmetric key specific to each infraEnv. This token will be used
+as authentication for requests to assisted service, specifically for the endpoints
+required by the image service.
+
+## Motivation
+
+When the assisted service is deployed in the cloud, it relies on Red Hat
+SSO for user authentication and authentication tokens from Red Hat SSO are not
+usable with pre-signed URLs for two main reasons:
+
+* Size - these tokens are typically ~2000 characters long
+* Expiration - these tokens have a maximum lifespan of 15 minutes
+
+### Goals
+
+* Define an authentication strategy suitable for pre-signed URLs in the cloud. This includes a
+token with a longer expiration (several hours) and a reasonable size (10s to 100s of characters)
+which will be included directly in a download URL
+
+### Non-Goals
+
+* Tie an image download URL to a Red Hat account or organization
+* Maintain public/private key pairs as service configuration to generate and issue tokens
+* Add additional external identity management components
+
+## Proposal
+
+The new authentication mechanism will be a JWT signed by the assisted service
+using a randomly generated key stored with the infraEnv. The JWT will include
+an `exp` claim as defined in RFC7519. A token will be included as a parameter
```

Comment:
Will we also include a `sub` claim identifying the infraenv? I don't think it would strictly be required, since the signature won't be valid unless using the token is used to access the matching infraenv, but it seems reasonable for observability to include an identifier in the JWT.

Reply:
Not adding it saves about 60 characters in the token. I guess it depends on how much we care about that length.